Firebase/Auth/Source/FIRAuth.m

```diff
@@ -1086,10 +1092,15 @@ - (void)phoneNumberSignInWithRequest:(FIRVerifyPhoneNumberRequest *)request
 }
 #endif
 
-- (void)notifyListenersOfAuthStateChangeWithUser:(FIRUser *)user token:(NSString *)token {
-  if (user != _currentUser) {
+/** @fn possiblyPostAuthStateChangeNotification
+    @brief Posts the auth state change notificaton if current user's token has been changed.
+ */
+- (void)possiblyPostAuthStateChangeNotification {
+  NSString *token = _currentUser.rawAccessToken;
+  if (_lastNotifiedUserToken == token || [_lastNotifiedUserToken isEqualToString:token]) {
```

Comment:
Do we care if the pointers are the same in this condition? I am inclined to think just the checking one string's content vs the other may be enough here.

Reply:
`isEqualToString:` (or `isEqual:`) returns incorrect result if both objects are `nil`, thus we need the extra pointer check here.

Reply:
I see. Makes sense, thanks.